### DISCUSSION TITLE
playing with audio settings

### DISCUSSION THREAD

**Discussion by airigami** [mcguirepr89/BirdNET-Pi#334] [Q&A]
I'm very confused about selecting an audio device. I have an electret mic assembled based on discussions here. I have it plugged into a UGreen adapter. I'm not happy with the results I'm seeing. I'm trying to play with ALSA settings, which looks like I need to select an audio card rather than using "default". But anything I enter other than "default" gets me an error opening the device. I'm not sure what I should be entering.

$ play -L

null
    Discard all samples (playback) or generate zero samples (capture)
default
    Playback/recording through the PulseAudio sound server
lavrate
    Rate Converter Plugin Using Libav/FFmpeg Library
samplerate
    Rate Converter Plugin Using Samplerate Library
speexrate
    Rate Converter Plugin Using Speex Resampler
jack
    JACK Audio Connection Kit
oss
    Open Sound System
pulse
    PulseAudio Sound Server
upmix
    Plugin for channel upmix (4,6,8)
vdownmix
    Plugin for channel downmix (stereo) with a simple spacialization
hw:CARD=Headphones,DEV=0
    bcm2835 Headphones, bcm2835 Headphones
    Direct hardware device without any conversions
plughw:CARD=Headphones,DEV=0
    bcm2835 Headphones, bcm2835 Headphones
    Hardware device with all software conversions
sysdefault:CARD=Headphones
    bcm2835 Headphones, bcm2835 Headphones
    Default Audio Device
dmix:CARD=Headphones,DEV=0
    bcm2835 Headphones, bcm2835 Headphones
    Direct sample mixing device
usbstream:CARD=Headphones
    bcm2835 Headphones
    USB Stream Output
hw:CARD=Device,DEV=0
    USB Audio Device, USB Audio
    Direct hardware device without any conversions
plughw:CARD=Device,DEV=0
    USB Audio Device, USB Audio
    Hardware device with all software conversions
sysdefault:CARD=Device
    USB Audio Device, USB Audio
    Default Audio Device
front:CARD=Device,DEV=0
    USB Audio Device, USB Audio
    Front output / input
surround21:CARD=Device,DEV=0
    USB Audio Device, USB Audio
    2.1 Surround output to Front and Subwoofer speakers
surround40:CARD=Device,DEV=0
    USB Audio Device, USB Audio
    4.0 Surround output to Front and Rear speakers
surround41:CARD=Device,DEV=0
    USB Audio Device, USB Audio
    4.1 Surround output to Front, Rear and Subwoofer speakers
surround50:CARD=Device,DEV=0
    USB Audio Device, USB Audio
    5.0 Surround output to Front, Center and Rear speakers
surround51:CARD=Device,DEV=0
    USB Audio Device, USB Audio
    5.1 Surround output to Front, Center, Rear and Subwoofer speakers
surround71:CARD=Device,DEV=0
    USB Audio Device, USB Audio
    7.1 Surround output to Front, Center, Side, Rear and Woofer speakers
iec958:CARD=Device,DEV=0
    USB Audio Device, USB Audio
    IEC958 (S/PDIF) Digital Audio Output
dmix:CARD=Device,DEV=0
    USB Audio Device, USB Audio
    Direct sample mixing device
usbstream:CARD=Device
    USB Audio Device
    USB Stream Output
hw:CARD=vc4hdmi0,DEV=0
    vc4-hdmi-0, MAI PCM i2s-hifi-0
    Direct hardware device without any conversions
plughw:CARD=vc4hdmi0,DEV=0
    vc4-hdmi-0, MAI PCM i2s-hifi-0
    Hardware device with all software conversions
sysdefault:CARD=vc4hdmi0
    vc4-hdmi-0, MAI PCM i2s-hifi-0
    Default Audio Device
hdmi:CARD=vc4hdmi0,DEV=0
    vc4-hdmi-0, MAI PCM i2s-hifi-0
    HDMI Audio Output
dmix:CARD=vc4hdmi0,DEV=0
    vc4-hdmi-0, MAI PCM i2s-hifi-0
    Direct sample mixing device
usbstream:CARD=vc4hdmi0
    vc4-hdmi-0
    USB Stream Output
hw:CARD=vc4hdmi1,DEV=0
    vc4-hdmi-1, MAI PCM i2s-hifi-0
    Direct hardware device without any conversions
plughw:CARD=vc4hdmi1,DEV=0
    vc4-hdmi-1, MAI PCM i2s-hifi-0
    Hardware device with all software conversions
sysdefault:CARD=vc4hdmi1
    vc4-hdmi-1, MAI PCM i2s-hifi-0
    Default Audio Device
hdmi:CARD=vc4hdmi1,DEV=0
    vc4-hdmi-1, MAI PCM i2s-hifi-0
    HDMI Audio Output
dmix:CARD=vc4hdmi1,DEV=0
    vc4-hdmi-1, MAI PCM i2s-hifi-0
    Direct sample mixing device
usbstream:CARD=vc4hdmi1
    vc4-hdmi-1
    USB Stream Output

If I pick another device, and then look at the log:

$ journalctl -fu birdnet_recording.service -ulivestream.service

May 29 09:55:22 birdnet1 systemd[1]: livestream.service: Scheduled restart job, restart counter is at 24.
May 29 09:55:22 birdnet1 systemd[1]: Stopped BirdNET-Pi Live Stream.
May 29 09:55:22 birdnet1 systemd[1]: Started BirdNET-Pi Live Stream.
May 29 09:55:22 birdnet1 livestream.sh[2940136]: ffmpeg version 4.3.4-0+deb11u1+rpt1 Copyright (c) 2000-2021 the FFmpeg developers
May 29 09:55:22 birdnet1 livestream.sh[2940136]:   built with gcc 10 (Debian 10.2.1-6)
May 29 09:55:22 birdnet1 livestream.sh[2940136]:   configuration: --prefix=/usr --extra-version=0+deb11u1+rpt1 --toolchain=hardened --incdir=/usr/include/aarch64-linux-gnu --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-mmal --enable-neon --enable-v4l2-request --enable-libudev --enable-epoxy --enable-sand --libdir=/usr/lib/aarch64-linux-gnu --arch=arm64 --enable-pocketsphinx --enable-libdc1394 --enable-libdrm --enable-vout-drm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
May 29 09:55:22 birdnet1 livestream.sh[2940136]:   libavutil      56. 51.100 / 56. 51.100
May 29 09:55:22 birdnet1 livestream.sh[2940136]:   libavcodec     58. 91.100 / 58. 91.100
May 29 09:55:22 birdnet1 livestream.sh[2940136]:   libavformat    58. 45.100 / 58. 45.100
May 29 09:55:22 birdnet1 livestream.sh[2940136]:   libavdevice    58. 10.100 / 58. 10.100
May 29 09:55:22 birdnet1 livestream.sh[2940136]:   libavfilter     7. 85.100 /  7. 85.100
May 29 09:55:22 birdnet1 livestream.sh[2940136]:   libavresample   4.  0.  0 /  4.  0.  0
May 29 09:55:22 birdnet1 livestream.sh[2940136]:   libswscale      5.  7.100 /  5.  7.100
May 29 09:55:22 birdnet1 livestream.sh[2940136]:   libswresample   3.  7.100 /  3.  7.100
May 29 09:55:22 birdnet1 livestream.sh[2940136]:   libpostproc    55.  7.100 / 55.  7.100
May 29 09:55:22 birdnet1 livestream.sh[2940136]: Trailing option(s) found in the command: may be ignored.
May 29 09:55:22 birdnet1 livestream.sh[2940136]: [alsa @ 0x55987ef050] cannot open audio device hw:CARD=Device (Device or resource busy)
May 29 09:55:22 birdnet1 livestream.sh[2940136]: hw:CARD=Device: Input/output error
May 29 09:55:22 birdnet1 systemd[1]: livestream.service: Main process exited, code=exited, status=1/FAILURE
May 29 09:55:22 birdnet1 systemd[1]: livestream.service: Failed with result 'exit-code'.

What should I be using?

Thanks.

**Comment by mcguirepr89** [comment]
Hi, @airigami 
You should definitely be using "`default`" -- default means "PulseAudio" and pulseaudio is a wrapper for ALSA devices.

So, you can adjust ALSA settings AND keep the Rec Card setting to "default".

To adjust the settings, [follow this guide](https://github.com/mcguirepr89/BirdNET-Pi/wiki/Adjusting-your-sound-card). Pulse is going to use the correct device, so you really don't need to change the sound card in "Advanced Settings". You _will_ want to change the "Channels" to 1 if your input is mono.

**Reply by airigami** [reply]
I have channels on 1. I started looking further because playing with ALSA settings isn't really making a difference. Thought I was missing something. The electrets I'm using are these:
    https://www.digikey.com/en/products/detail/pui-audio-inc/AOM-5024L-HD-F-R/12152286?utm_medium=email&utm_source=oce&utm_campaign=4251_OCE22RT&utm_content=productdetail_US&utm_cid=2841574&so=75678878&mkt_tok=MDI4LVNYSy01MDcAAAGEL0_AJnXafoIfkppOUIl4jstzMiVtWK-76L3XTlUhGjMqfcnQI97g0649USpSc9nQe-Ta8A3_sq1Qgh4LJw12rfmPu7PXps87s9RnUZtR

I'm getting a lot of low frequency noise, and it's not very sensitive. With a nice condenser mic, I was getting much better recordings and much higher confidence. I expect to get better results with an expensive mic, but after seeing the difference, I'm hoping to improve things without leaving the expensive mics on there.

**Reply by mcguirepr89** [reply]
Perhaps @nilspupils or @DD4WH can ask the right questions to discern why your electret build is not giving you the same results we've seen on other builds/installations?

**Reply by DD4WH** [reply]
@airigami 
please give us a close-up photgraph of your mic soldering and a close up photo of your soldering of the cable to the 3.5mm plug. Then we can help you solve the solder issue of your capsule.
Here you find some help regarding connections:

https://github.com/mcguirepr89/BirdNET-Pi/wiki/Microphones-and-USB-audio-sound-cards-for-BirdNET-Pi

**Reply by mcguirepr89** [reply]
Thank you, @DD4WH!!!!!!!!

**Reply by nilspupils** [reply]
Hi @airigami, it is not clear if you used "Alsamixer" and if you adjusted the volume settings there. You should set the settings both in default (Pulseaudio) and your USB-soundcard. Can you confirm you did that?
Best, 
Nils

**Reply by airigami** [reply]
hmm. I have to cut apart the shrink tubing to show the soldering. I guess I can do that. I had a 3.5mm TRS audio cable sitting around that I cut in half, so I didn't solder the plug end. ok, looking at the diagram you just included, I see that has TR tied together. I didn't do that. I only have the red wire going to tip. Actually that diagram is helpful. I'll start there. Thanks.

**Reply by airigami** [reply]
Yes, I did adjust the level is Pulseaudio and the card.

**Reply by nilspupils** [reply]
Great, could you also tell me what UGREEN device you use, as there are several different ones.

**Reply by airigami** [reply]
Would be great if I could find a model number. I used this one since it puts it at a distance from the Pi without an extension cable: https://www.amazon.com/dp/B01N905VOY?ref=ppx_yo2ov_dt_b_product_details&th=1

**Reply by nilspupils** [reply]
Ok, this one should work with your TRS plug. Then a problem with the connections is likely. Can you provide an image of the soldering?

**Reply by airigami** [reply]
![IMG_3675](https://user-images.githubusercontent.com/10488194/170891221-a96d98f0-c16d-48d2-8be6-9caa06116131.jpeg)

There's the solder connection. The 3rd wire is just floating. I can solder that to the positive connection on the mic also if that will make a difference.

**Reply by nilspupils** [reply]
It's hard to see, but please make sure none of the cables touches the case of the capsule. The case is grounded and you will have a short when the + cable touches the case. Can you confirm the capsule was only soldered for a short time and didn't overheat?

**Reply by airigami** [reply]
hmm. I know I tested it before I sealed it up. But, I can verify now that I have continuity between all three: case, +, -. I'll start over. I have more. I also cleaned up my soldering station and found some unused tips, so I won't be fighting with a blackened tip that "someone" ruined.

**Reply by nilspupils** [reply]
Then good luck and let us know if it works out.

**Reply by airigami** [reply]
Starting over with a new electret made a huge difference. Still not what I was getting with a condenser mic, but it’s confidence scores are up now and the matches are far better.  I had a lot of misidentifications with my messed up mic. 

On the topic of misindentifications, I know I saw discussions about making it easy to delete those. Is that something coming soon? I’m yet to capture a screech owl for real (we do have them around here so I don’t want to prevent that from being captured) but every police/ambulance siren gets identified that way.

**Reply by ehpersonal38** [reply]
@airigami It would be pretty easy to implement a "delete" feature for recordings, I'll look into that. 

Also, could you tell me the average confidence of the screech owl misidentifications? I was thinking maybe if a police siren always gets picked up as a certain confidence (say 0.5-0.6), we could filter those out before they get detected. But if the confidence is all over the place then that idea is out. So do let me know! 🙂

**Reply by airigami** [reply]
For an average, I'll have to let it run for a few days with the new mic and see what I get. The crappy mic was giving me confidence scores for sirens/screech owls at everywhere from 75-85%. With the better mic on it last night, I saw a couple 94-95%. I'm pretty sure with the even nicer mics I played with, my scores were up there as well.

**Comment by DougWelch** [comment]
In my considerable experience, sirens will be detected as Eastern 
Screech Owl detections with the whole range of confidences. The only 
real solution is to train a model with sirens identified as not birds. 3 
kHz backup alarms do the same thing for Northern Saw-whet Owl detections!

Cheers,
Doug
